src/routes/home.tsx

```diff
@@ -5,27 +5,55 @@ import Spinner from '@/components/spinner'
 import { compareEvents } from '@/utils/compare-events'
 import { useGeoLocation } from '@/utils/use-geo-location'
 import { useEvents } from '@/cache/events/use'
+import { useState } from 'preact/hooks'
 
 const now = new Date()
-
 const Home = () => {
   const events = useEvents()
   const location = useGeoLocation()
+  const [term = '', setTerm] = useState(0);
+
+  function searchingFor(term: string | number) {
+    return function (x: any) {
```

Comment:
I'd suggest not using the `any` type, because we can use a more accurate type. Also renaming the variable would be helpful.

```js
    return function (event: ProcessedEventInfo) {
```

You will also need to add:

```js
import { ProcessedEventInfo } from '@/api/event-info'
```

Reply:
When I tried to  defined type event: ProccessedEventInfo. There is a problem with properties district and fulldistrict. This error occurred. App  running correctly anyway, but i cannot figure out how to solve this issue . Do you have any suggestion? 
![event](https://user-images.githubusercontent.com/55954157/67437031-31532700-f5f0-11e9-8b86-14db178fb615.png)

Reply:
Yeah. I think if you do:

```js
if (
  event.district !== undefined &&
  event.district.toLowerCase().includes(lowercaseQuery)
) {
  return true;
}

if (
  event.fullDistrict !== undefined &&
  event.fullDistrict.toLowerCase().includes(lowercaseQuery)
) {
  return true;
}
```

Then that should hopefully fix the type errors and also that allows us to not have to duplicate checking name/key/locationName (because it only returns if it matches)

Reply:
Yes it works. Thanks for advice